### PR TITLE
Finished the homework 8

### PR DESCRIPTION
**Pull request by MaximIurceac**
Please check my hw

### CODE REVIEW

File: src/main/java/org/example/homework_nr_8/Animal.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package org.example.homework_nr_8;
+
+public interface Animal {
+    void eat();
+    void run();
+    default void drink(){
```
Comment (gh3rtzz) [inline src/main/java/org/example/homework_nr_8/Animal.java:6]: add a simple implementation here. If you want it to do nothing, make it print out to console that it's not overriden.

**Comment by MaximIurceac** [comment]
I added the implementation

File: src/main/java/org/example/homework_nr_8/Elephant.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    @Override
+    public void run() {
+        System.out.println("Elephant is running");
+
+    }
+
+    @Override
+    public void drink() {
+        Animal.super.drink();
```
Comment (gh3rtzz) [inline src/main/java/org/example/homework_nr_8/Elephant.java]: Clever, but, it voids the usage of the default methods in interfaces.

Picture this:

You have a default method in an interface. By interface rules default method must have a body
One more rule is that default methods do not have to be overriden if we compare them to abstract methods.
Therefore, if you override the default method and then call super of the interface and call the default method, it doesn't make sense.

When you override it here, create a new implementation.
ex. `@Override
public void drink(){
System.out.println("Elephant is drinking");`

File: src/main/java/org/example/homework_nr_8/Elephant.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    @Override
+    public void sleep() {
+        System.out.println("Erbivor is sleeping");
```
Comment (gh3rtzz) [inline src/main/java/org/example/homework_nr_8/Elephant.java]: Why is it erbivor? You're overriding this method inside the Elephant class and not Erbivor

**Comment by MaximIurceac** [comment]
Tell me pls. It will be right if I delete the method "drink" in Elephant class and left the method in interface "Animal"?
I watched the leasson again and you are right that the point of default class is lost.

**Review by gh3rtzz — CHANGES_REQUESTED** [review]
Use the default method drink() somewhere

File: src/main/java/org/example/homework_nr_8/Elephant.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    public static void main(String[] args) {
+        Animal elephant = new Elephant();
+        Erbivor elephant1 = new Elephant();
+
+
+
+
+        elephant.eat();
+        elephant.run();
+        elephant.drink();
```
Comment (MaximIurceac) [inline src/main/java/org/example/homework_nr_8/Elephant.java:14]: I'm using it here and it works

**Review by gh3rtzz — APPROVED** [review]
approved

**Pull request merged**